torch/distributions/chi2.py

```diff
@@ -19,5 +19,6 @@ class Chi2(Gamma):
     """
     params = {'df': constraints.positive}
 
-    def __init__(self, df):
-        super(Chi2, self).__init__(0.5 * df, 0.5)
+    def __init__(self, df, validate_args=None):
+        self.df = df
```

Comment:
This might cause issues, since `df` is not broadcasted.

Reply:
That's right, and I think we shouldn't be saving `df`. Instead we should reconstruct it as needed
```py
@property
def df(self):
    return 2 * self.concentration
```
This minimizes overhead when `self._validate_args == False`. It is more expensive when `self._validate_args == True`, but in that's not the performance-critical case anyway.

Reply:
@lazypanda1 I have taken care of this in my branch. I think once you are cherry-picked those commits from there, some of these issues might be resolved.

Reply:
Please implement `.df` as a property as suggested.